src/api/java/com/minecolonies/api/util/ItemStackUtils.java

```diff
@@ -991,5 +988,41 @@ public static ItemStack idToItemStack(final String itemData)
         }
         return stack;
     }
+
+    /**
+     * Obtains a list of all basic items in the game, plus any extra items present in the player's
+     * inventory (allowing for items with custom NBT, e.g. DO blocks or dyed armour).
+     *
+     * @param player The player whose inventory to check.
+     * @return The set of items.
+     */
+    public static Set<ItemStack> allItemsPlusInventory(@NotNull final PlayerEntity player)
+    {
+        // using ItemStorage for better hashing
+        final Set<ItemStorage> allItems = new HashSet<>();
+
+        // all known items
+        for (final ItemStack stack : IColonyManager.getInstance().getCompatibilityManager().getListOfAllItems())
```

Comment:
I feel like we should partially cache this or so. So it's not rebuilt on every search

Reply:
Done.